sentry/src/main/java/io/sentry/protocol/TransactionNameSource.java

```diff
@@ -0,0 +1,104 @@
+package io.sentry.protocol;
+
+import io.sentry.ILogger;
+import io.sentry.JsonDeserializer;
+import io.sentry.JsonObjectReader;
+import io.sentry.JsonObjectWriter;
+import io.sentry.JsonSerializable;
+import java.io.IOException;
+import java.util.Locale;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public enum TransactionNameSource implements JsonSerializable {
```

Comment:
This is now an `enum` which is good, what if this SDK ser/deser objects from Hybrid SDKs, they might have sources that this SDK does not know yet, wondering if this should be a String to make it flexible for this use case.
Maybe @bruno-garcia has ideas here.

Reply:
We could use String in TransactionInfo like this: https://github.com/getsentry/sentry-java/pull/2194
Would you prefer that?

Reply:
Merged the other PRs in here now as seems more stable to me to rely on String based de/serialization